lib/client/presence.go

```diff
@@ -23,26 +23,59 @@ import (
 	"time"
 
 	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
 
 	"github.com/gravitational/teleport/api/client/proto"
-	"github.com/gravitational/teleport/lib/auth"
 )
 
-func runPresenceTask(ctx context.Context, term io.Writer, auth auth.ClientI, tc *TeleportClient, sessionID string) error {
+// PresenceMaintainer allows maintaining presence with the Auth service.
+type PresenceMaintainer interface {
+	MaintainSessionPresence(ctx context.Context) (proto.AuthService_MaintainSessionPresenceClient, error)
+}
+
+const mfaChallengeInterval = time.Second * 30
+
+// PresenceOptions allows passing optional overrides
+// to RunPresenceTask. Mainly used by tests.
+type PresenceOptions struct {
```

Comment:
Maybe unexport this to keep the API surface smaller? (Other packages only use the exported `WithXXX` functions and don't need to know about the internal struct)